python/pyarrow/tests/test_dataset.py

```diff
@@ -1102,23 +1092,14 @@ def test_fragments_parquet_subset_ids(tempdir):
     # select with row group ids
     subfrag = fragment.subset(row_group_ids=[0, 3])
     assert subfrag.num_row_groups == 2
-    # the row_groups list is initialized, but don't have statistics
-    assert len(subfrag.row_groups) == 2
-    assert subfrag.row_groups[0].statistics is None
+    assert subfrag.row_groups == [fragment.row_groups[i] for i in [0, 3]]
+
     # check correct scan result of subset
     result = subfrag.to_table()
     assert result.to_pydict() == {"f1": [0, 3], "f2": [1, 1]}
 
-    # if the original fragment has statistics -> preserve them
-    fragment.ensure_complete_metadata()
-    subfrag = fragment.subset(row_group_ids=[0, 3])
-    assert subfrag.num_row_groups == 2
-    assert len(subfrag.row_groups) == 2
-    assert subfrag.row_groups[0].statistics is not None
-
     # empty list of ids
     subfrag = fragment.subset(row_group_ids=[])
-    assert subfrag.num_row_groups == 0
```

Comment:
This is something we can keep? (it should still be true, and the `num_row_groups` is independent from `.row_groups` on the python side)